### ISSUE TITLE
100% cpu on one core with files with many comments

### ISSUE THREAD

**Issue by Ddystopia** [nvim-treesitter/nvim-treesitter#4763]
### Describe the bug

When I'm opening large files with a large amount of comments, and amount of not commented code could be very small, in insert mode while typing characters one cpu thread goes to 100% and it lags very strongly. If file is made mostly of comments, it should not be a hard task to handle it.

### To Reproduce

Use that [configuration](https://drive.google.com/file/d/14JCCvSFFMiXnRCN4qYN2lWBnZ7ZP1rq_/view?usp=share_link)
My config structure:
![image](https://user-images.githubusercontent.com/63151578/236513255-c9e204a3-8e17-41f0-9d27-f136db183691.png)
Contents of these files are in `Additional context` section and by link above.

Open [this file](https://drive.google.com/file/d/1EUa07SmDP02OCNgmrg11aS-A1Xo40uRn/view?usp=share_link).
I've opened it like that: `XDG_CONFIG_HOME=~/nvimtest nvim nservice.js`

Wait untill "Lazy" package manager will be installed, screen will be blank for a moment.

Run `:TSInstall javascript`

Try to edit file, for example type "example" at the beginning. It will lug.
If you'll run `:TSDisable highlight` everything will work as it should.

### Expected behavior

Everything should be smooth and not take up so many resources.
If file is made mostly of comments, it should not be a hard task to handle it.

### Output of `:checkhealth nvim-treesitter`

```markdown
nvim: require("nvim.health").check()

Configuration ~
- OK no issues found

Runtime ~
- OK $VIMRUNTIME: /usr/share/nvim/runtime

Performance ~
- OK Build type: Release

Remote Plugins ~
- OK Up to date

terminal ~
- key_backspace (kbs) terminfo entry: `key_backspace=\177`
- key_dc (kdch1) terminfo entry: `key_dc=\E[3~`
- $COLORTERM="truecolor"
```

### Output of `nvim --version`

```text
> nvim --version
NVIM v0.9.0
Build type: Release
LuaJIT 2.1.0-beta3

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

Run :checkhealth for more info
```

### Additional context

Config:
init.lua
```lua
local lazypath = '/home/ddystopia/nvimtest/data/lazy/lazy.nvim'
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({
    'git', 'clone', '--filter=blob:none', '--single-branch',
    'https://github.com/folke/lazy.nvim.git', lazypath
  })
end
vim.opt.runtimepath:prepend(lazypath)

require('lazy').setup('plugins', {
})
```
lua/plugins.lua
```lua
return {
  'nvim-treesitter/nvim-treesitter',
  event = { 'BufReadPost' },
  config = function()
    local treesitter = require('nvim-treesitter.configs')

    treesitter.setup {
      highlight = { enable = true },
    }
  end
}
```

**Comment by clason** [comment]
That's a known issue with many injections (which comments are, if you have installed the comment parser -- you haven't shared an _actual_ minimal reproducing config; nobody will download and test your full setup!) that is actively being worked on in Neovim. Try the latest master, or uninstall the comment parser.

**Comment by Ddystopia** [comment]
This is not my full setup, this there is only 2 files - with package manager and treesitter

**Comment by Ddystopia** [comment]
> Yes. This does not have _any_ parsers installed, so if you are seeing the issue with _these_ files (only), it has nothing to do with treesitter.

But when treesitter is disabled, everything works as it should. How can it has nothing to do with treesitter?

**Comment by clason** [comment]
That's why I'm saying that the two files are not enough to reproduce the issue.

**Comment by Ddystopia** [comment]
How can I improve my issue?

**Comment by clason** [comment]
Add a complete config that I can copy and use to reproduce the issue.

**Comment by Ddystopia** [comment]
@clason Thank you for an advise. There is an updated minimal configuration.
File: https://drive.google.com/file/d/1EUa07SmDP02OCNgmrg11aS-A1Xo40uRn/view?usp=share_link
Configuration:
```lua
local ensure_packer = function()
  local fn = vim.fn
  local install_path = fn.stdpath('data')..'/site/pack/packer/start/packer.nvim'
  if fn.empty(fn.glob(install_path)) > 0 then
    fn.system({'git', 'clone', '--depth', '1', 'https://github.com/wbthomason/packer.nvim', install_path})
    vim.cmd [[packadd packer.nvim]]
    return true
  end
  return false
end

local packer_bootstrap = ensure_packer()

require('packer').startup(function(use)
  use 'wbthomason/packer.nvim'
  use 'nvim-treesitter/nvim-treesitter'

  local treesitter = require('nvim-treesitter.configs')

  treesitter.setup {
    highlight = { enable = true },
  }

  if packer_bootstrap then
    require('packer').sync()
  end

end)

```

**Comment by Ddystopia** [comment]
@clason Sorry for the inconvenience, but I can't reproduce on other machines myself. I try different versions of neovim and so on. Do you know what might be causing this? On my system with this config I have very noticeable brakes when editing text, on others, even on virtual machines on the same computer everything is fine. Thanks in advance

**Comment by clason** [comment]
No, sorry, I have no idea why this might happen, given the information.
